src/main/java/com/questhelper/helpers/quests/pryingtimes/PryingTimes.java

```diff
@@ -0,0 +1,215 @@
+ */
+package com.questhelper.helpers.quests.pryingtimes;
+
+import com.questhelper.collections.ItemCollections;
+import com.questhelper.panel.PanelDetails;
+import com.questhelper.questhelpers.BasicQuestHelper;
+import com.questhelper.questinfo.QuestHelperQuest;
+import com.questhelper.requirements.Requirement;
+import com.questhelper.requirements.item.ItemRequirement;
+import com.questhelper.requirements.item.TeleportItemRequirement;
+import com.questhelper.requirements.npc.NpcRequirement;
+import com.questhelper.requirements.player.FreePortTaskSlotsRequirement;
+import com.questhelper.requirements.player.ShipInPortRequirement;
+import com.questhelper.requirements.player.SkillRequirement;
+import com.questhelper.requirements.quest.QuestRequirement;
+import com.questhelper.requirements.util.Port;
+import com.questhelper.requirements.var.VarbitRequirement;
+import com.questhelper.requirements.zone.Zone;
+import com.questhelper.rewards.ExperienceReward;
+import com.questhelper.rewards.ItemReward;
+import com.questhelper.rewards.QuestPointReward;
+import com.questhelper.rewards.UnlockReward;
+import com.questhelper.steps.ConditionalStep;
+import com.questhelper.steps.ItemStep;
+import com.questhelper.steps.NpcStep;
+import com.questhelper.steps.ObjectStep;
+import com.questhelper.steps.PortTaskStep;
+import com.questhelper.steps.QuestStep;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+import com.questhelper.steps.SailStep;
+import net.runelite.api.QuestState;
+import net.runelite.api.Skill;
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.api.gameval.ItemID;
+import net.runelite.api.gameval.NpcID;
+import net.runelite.api.gameval.VarbitID;
+import static com.questhelper.requirements.util.LogicHelper.and;
+import static com.questhelper.requirements.util.LogicHelper.not;
+
+public class PryingTimes extends BasicQuestHelper
+{
+	ItemRequirement hammerRequirement, steelBarRequirement, redberryPieRequirement, captainsLogRequirement, gotTheKey, gotStout;
+	SkillRequirement sailingSkillRequirement, smithingSkillRequirement;
+	QuestRequirement pandemoniumQuestRequirement, knightsSwordQuestRequirement;
+	FreePortTaskSlotsRequirement freeTaskSlotRequirement;
+	ShipInPortRequirement shipAtPortSarimDock;
+	TeleportItemRequirement thurgoTeleportRecommend;
+	NpcRequirement spawnedTroll;
+	VarbitRequirement drankStout;
+
+	Zone portSarimDockZone;
+
+	NpcStep startQuest, letSteveKnow, getKey, giveKey, killTheTroll, goToSteve;
+	PortTaskStep deliverCargo;
+	ObjectStep testKey, openCrate;
+	SailStep sailToCrate;
+	ItemStep drinkTheStout;
+
+	@Override
+	public Map<Integer, QuestStep> loadSteps()
+	{
+		initializeRequirements();
+		setupConditions();
+		setupSteps();
+		Map<Integer, QuestStep> steps = new HashMap<>();
+		steps.put(0,  startQuest);
+		steps.put(5,  deliverCargo);
```

Comment:
Step 5 should probably be a conditional: I have a feeling you can remove the task, but I do not know what the option to get it again from Steve is. 
```suggestion
        ConditionalStep cDeliverCargo = new ConditionalStep(getTaskFromSteve)
        cDeliverCargo.addStep(hasTask600, deliverCargo)
		steps.put(5,  cDeliverCargo);
```